batch/batch/worker/worker.py

```diff
@@ -178,6 +180,77 @@ def compose(auth: Union[MutableMapping, str, bytes], registry_addr: str = None):
 image_lock = aiorwlock.RWLock()
 
 
+async def read_file(fs: LocalAsyncFS, path: str) -> str:
+    if os.path.exists(path):
+        stream = await fs.open(path)
+        async with stream:
+            return (await stream.read()).decode()
+    return ''
+
+
+class ResourceUsageMonitor:
+    VERSION = 1
+
+    def __init__(self, container_name: str, output_file_path: str):
+        self.container_name = container_name
+        self.output_file_path = output_file_path
+
+        self.last_time_nanosecs: Optional[int] = None
+        self.last_cpu_nanosecs: Optional[int] = None
+
+        self.task: Optional[asyncio.Future] = None
+
+    def write_header(self):
+        with open(self.output_file_path, 'w') as f:
+            f.write(f'# version={self.VERSION}' + '\n')
+            f.write('\t'.join(['time_msecs', 'cpu_usage', 'memory_usage_bytes']) + '\n')
+
+    def cpu_usage(self) -> Optional[float]:
+        now_time_nanosecs = time_nanosecs()
+        try:
+            with open(f'/sys/fs/cgroup/cpu/{self.container_name}/cpuacct.usage', 'r') as f:
+                now_cpu_nanosecs = int(f.read().rstrip())
+        except FileNotFoundError:
+            return None
+
+        if self.last_time_nanosecs is not None:
+            assert self.last_cpu_nanosecs is not None
+            cpu_usage = (now_cpu_nanosecs - self.last_cpu_nanosecs) / (now_time_nanosecs - self.last_time_nanosecs)
+        else:
+            cpu_usage = None
+
+        self.last_time_nanosecs = now_time_nanosecs
+        self.last_cpu_nanosecs = now_cpu_nanosecs
+
+        return cpu_usage
+
+    def memory_usage_bytes(self) -> Optional[int]:
+        try:
+            with open(f'/sys/fs/cgroup/memory/{self.container_name}/memory.usage_in_bytes', 'r') as f:
+                return int(f.read().rstrip())
+        except FileNotFoundError:
+            return None
```

Comment:
There's probably a race condition where there's a delay in the container being created and when we start measuring because we don't know when the container has actually started. All we know is we have started the process that calls `crun run`. Would you rather make another `crun` call to get whether the container exists yet before starting to measure?

FWIW, the FileNotFoundError happened every time.